pwa-devdocs/src/venia-pwa-concept/features/modular-components/index.md

```diff
@@ -0,0 +1,200 @@
+| `store`   | [Redux Store][]   | The application store to pass on to a [Redux `Provider`][] component       |
+| `apiBase` | `string`          | Root URL of the Magento store to use in the [Peregrine Router][] component |
+{:style="table-layout:auto"}
+
+### Venia drivers
+
+The [`src/drivers`][] dependency is a centralized module for Venia components that rely on external dependencies, such as GraphQL clients and Redux stores.
+Instead of importing these dependencies directly, Venia components import them from `src/drivers`.
+
+```js
+import { Link, resourceUrl } from 'src/drivers';
+```
+
+The default implementation, which is used in the Venia storefront, provides modules that work with the components provided by the [Venia Adapter][].
+
+| Module name   | Source             |
+| ------------- | ------------------ |
+| `Query`       | `react-apollo`     |
+| `Link`        | `react-router-dom` |
+| `Redirect`    | `react-router-dom` |
+| `Route`       | `react-router-dom` |
+| `resourceUrl` | [`makeUrl.js`][]   |
+| `Adapter`     | [`adapter.js`][]   |
+| `connect`     | `react-redux`      |
+{:style="table-layout:auto"}
+
+#### Custom drivers
+
+You can create a custom implementation of the `src/drivers` module to instead provide custom drivers for Venia components.
+This lets you use Venia components without wrapping them inside a Venia Adapter.
+
+Use a build tool, such as [Webpack][] or [Rollup][], to override driver module imports in Venia components.
+
+> Example `webpack.config.js`:
+
+```js
+module: {
+    alias: {
+      "@magento/venia-drivers": "./myReplacementDrivers"
+    }
+  }
+```
+
+> Example `./myReplacementDrivers`:
+
+```jsx
+import React, { Component } from 'react';
+import { resourceUrl as veniaResourceUrl } from '@magento/venia-concept/esm/drivers';
+
+// A replacement Query that loads forever
+export class Query extends Component {
+    render() {
+        return this.props.children({ loading: true });
+    }
+}
+
+// A replacement Link that doesn't use the client-side router
+export class Link extends Component {
+    render() {
+        const { children, to, ...other } = this.props;
+        return (
+            <a {...other} href={to}>
+                {children}
+            </a>
+        );
+    }
+}
+
+// A replacement resourceUrl that calls Venia's implementation by importing
+// Venia's default driver, then additionally validates urls and adds a parameter
+export function resourceUrl(...args) {
+    let url = veniaResourceUrl(...args);
+    try {
+        url = new window.URL(url);
+    } catch (e) {
+        url = new window.URL(url, window.location.origin);
+    }
+    const params = new URLSearchParams(url.search);
+    params.append('referrer', window.location.hostname);
+    url.search = `?${params}`;
+    return url.href;
+}
+
+// You can also override Router, Route, Redirect, and react-redux's
+// `connect()` HOC
```

Comment:
I noticed in @zetlen's example repo that he doesn't override these other functions. What happens if you write a custom adapter, and then import a venia component that requires something you haven't overwritten? Is there a suggested method for handling this scenario?